pkg/provider/aws/secret.go

```diff
@@ -0,0 +1,19 @@
+package aws
+
+import (
+	"github.com/klothoplatform/klotho/pkg/core"
+	"github.com/klothoplatform/klotho/pkg/provider/aws/resources"
+)
+
+func (a *AWS) GenerateSecretsResources(construct *core.Secrets, result *core.ConstructGraph, dag *core.ResourceGraph) error {
+	for _, single := range construct.Secrets {
+		secret := resources.NewSecret(construct.Provenance(), single, a.Config.AppName)
+		dag.AddResource(secret)
+		a.MapResourceDirectlyToConstruct(secret, construct)
+
```

Comment:
also does the execution units resource need to depend on the secret version? Because its an env var it likely does

Reply:
`MapResourceDirectlyToConstruct` will make the exec unit depend on the secret, right? And the secret depends on the secret-version, so the exec unit will transitively depend on the secret-version. Does it need to be a direct dependency for env vars?

Reply:
i guess it indirectly does because it will map as a dependency to the functions role and the function is dependent on that, so that may be fine yeah